lib/Dialect/Moore/MooreTypes.cpp

```diff
@@ -234,3 +240,12 @@ Type MooreDialect::parseType(DialectAsmParser &parser) const {
 void MooreDialect::printType(Type type, DialectAsmPrinter &printer) const {
   printMooreType(type, printer);
 }
+
+LogicalResult packedVerify(function_ref<InFlightDiagnostic()> emitError,
+                           ArrayRef<StructLikeMember> members) {
+  if (!llvm::all_of(members, [](const auto &member) {
+        return llvm::isa<PackedType>(member.type);
+      }))
+    return emitError() << "Struct/Union members must be packed types";
+  return success();
+}
```

Comment:
Can you make this a `static LogicalResult verifyAllMembersPacked(...)` function and move it directly before `StructType::verify`? That way you can remove the declaration further up in the file 🙂

Reply:
Thanks for your advice, I get your point.